File: app/src/main/java/com/regaan/lockroot/vault/VaultModels.kt
package com.regaan.lockroot.vault

import java.util.UUID

data class Vault(
    val vaultId: String = UUID.randomUUID().toString(),
    val schemaVersion: Int = 1,
    val entries: MutableList<VaultEntry> = mutableListOf(),
) {
    fun copyMutable(): Vault = copy(entries = entries.map { it.copy() }.toMutableList())
}

data class VaultEntry(
    val id: String = UUID.randomUUID().toString(),
    var title: String,
    var website: String,
    var username: String,
    var password: String,
    var notes: String,
    var tags: List<String> = emptyList(),
)

File: ios/Lockroot/Lockroot/Vault/VaultModels.swift
import Foundation

struct Vault: Codable, Equatable {
    var vaultId: String = UUID().uuidString
    var schemaVersion: Int = 1
    var entries: [VaultEntry] = []
}

struct VaultEntry: Codable, Identifiable, Equatable, Hashable {
    var id: String = UUID().uuidString
    var title: String
    var website: String
    var username: String
    var password: String
    var notes: String
    var tags: [String] = []

    static var empty: VaultEntry {
        VaultEntry(title: "", website: "", username: "", password: "", notes: "", tags: [])
    }
}
